### ISSUE TITLE
Downloading doesn't use trigger domain

### ISSUE THREAD

**Issue by mintopia** [tpill90/steam-lancache-prefill#14]
Currently the application makes connections directly to the hostnames of the CDN servers, relying on DNS poisoning of all CDN server hostnames to work correctly.

This needs to be updated to work in a way similar to steamcmd and the Steam Client, where it checks to see if `lancache.steamcontent.com` resolves to a private IP address. If so, it will open HTTP connections specifically to that IP address; and then use the `Host: ` header to specify the upstream CDN server; instead of connecting to the hostname of the CDN server.

We are planning to remove all the other Steam hostnames from uklans/cache-domains and only have the trigger domain, `lancache.steamcontent.com` at some point this year, so this behaviour will need updating before then.

**Comment by tpill90** [comment]
I tried switching over to using the trigger domain, with adding the host as the header.  I'm attaching a build with those changes here, for testing. 

Here's the actual change, does this look about right to you?
```csharp
 var url = ZString.Format("http://lancache.steamcontent.com/depot/{0}/chunk/{1}", request.DepotId, request.ChunkId);
 using var requestMessage = new HttpRequestMessage(HttpMethod.Get, url);
 requestMessage.Headers.Host = cdnServer.Host;
```

 [Link to the changes in context](https://github.com/tpill90/steam-lancache-prefill/blob/0dfd5435b530498f8691c878f09e1117fa0e6750/SteamPrefill/Handlers/DownloadHandler.cs#L88)

[linux-x64.zip](https://github.com/tpill90/steam-lancache-prefill/files/9154959/linux-x64.zip)
[win-x64.zip](https://github.com/tpill90/steam-lancache-prefill/files/9154961/win-x64.zip)

**Comment by tpill90** [comment]
Fixed in #16
